pkg/volume/csi/csi_mounter_test.go

```diff
@@ -694,11 +696,59 @@ func TestMounterSetUpWithFSGroup(t *testing.T) {
 			setFsGroup: true,
 			fsGroup:    3000,
 		},
+		{
+			name: "fstype, fsgroup, RWO provided, SupportsFSGroup Evaluate (should apply fsgroup)",
+			accessModes: []api.PersistentVolumeAccessMode{
+				api.ReadWriteOnce,
+			},
+			fsType:                "ext4",
+			setFsGroup:            true,
+			fsGroup:               3000,
+			driverSupportsFSGroup: true,
+			supportMode:           storage.EvaluateSupportsFSGroup,
+		},
+		{
+			name: "fstype, fsgroup, RWO provided, readonly, SupportsFSGroup Evaluate (should not apply fsgroup)",
+			accessModes: []api.PersistentVolumeAccessMode{
+				api.ReadWriteOnce,
+			},
+			readOnly:              true,
+			fsType:                "ext4",
+			setFsGroup:            true,
+			fsGroup:               3000,
+			driverSupportsFSGroup: true,
+			supportMode:           storage.EvaluateSupportsFSGroup,
+		},
+		{
+			name: "fstype, fsgroup, RWO provided, SupportsFSGroup Never (should not apply fsgroup)",
+			accessModes: []api.PersistentVolumeAccessMode{
+				api.ReadWriteOnce,
+			},
+			fsType:                "ext4",
+			setFsGroup:            true,
+			fsGroup:               3000,
+			driverSupportsFSGroup: true,
+			supportMode:           storage.NeverSupportsFSGroup,
+		},
+		{
+			name: "fstype, fsgroup, RWO provided, readOnly, SupportsFSGroup Always (should apply fsgroup)",
```

Comment:
What about a test case where fsgroup is not set?

And also more test cases with conditions where Evaluate wouldn't apply, like fstype, RWO

Reply:
I've included additional tests here, such as:

* default fstype with no fsgroup
* default type with fsgroup
* fstype and fsgroup specified, but ReadWriteMany AccessMode